compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -5,9 +5,12 @@
 //! used between functions, and they operate in a purely top-down
 //! way. Therefore, we break lifetime name resolution into a separate pass.
 
+// ignore-tidy-filelength
+
```

Comment:
Instead of this, could we instead move some diagnostics related code to a `diagnostics.rs` file?